src/IDCodeValidator.java

```diff
@@ -0,0 +1,193 @@
+import java.util.*;
+
+public class IDCodeValidator {
+
+
+    public static boolean isCorrect(String idCode) {
+        System.out.println(idCode);
+        IDNumberData personalCode = IDNumberData.getInformationFromIdCode(idCode);
+        if (idCode.length() == 11
+                && isGenderNumberCorrect(personalCode.genderNumber)
+                && isDayNumberCorrect(personalCode.dayNumber)
+                && isMonthNumberCorrect(personalCode.monthNumber)
+                && isBirthDateCorrect(personalCode.yearNumber, personalCode.monthNumber, personalCode.dayNumber)
+                && checkControlNumber(idCode)) {
+
+            System.out.println("ID is correct");
+            return true;
+        }
+        if (idCode.length() != 11) {
+            System.out.println("The length is not correct");
+
+        }
+
+        return false;
+    }
+
+
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (genderNumber == 1 || genderNumber == 3 || genderNumber == 5 || genderNumber == 2 || genderNumber == 4 || genderNumber == 6) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        boolean result;
+        int[] numbersOfDays = {0, 31, 28, 31, 30, 31, 30, 31, 30, 31, 30, 31};
+        if (numbersOfDays[2] == 28 && isLeapYear(yearNumber)) {
+            result = (dayNumber <= numbersOfDays[monthNumber] + 1);
+        } else {
+            result = (dayNumber <= numbersOfDays[monthNumber]);
+        }
+        if (result) {
+            System.out.println("Your birth date is correct");
+        } else {
+            System.out.println("Your birth date is not correct");
+        }
+        return result;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    public static boolean checkControlNumber(String idCode) {
+        int result = 0;
+        List<Integer> a = new ArrayList<>();
+        a.add(Integer.valueOf(idCode));
+        for (int i = 0; i < idCode.length(); i++) {
+            String n = String.valueOf(idCode.charAt(i));
+            a.add(Integer.parseInt(n));
+        }
+        List<Integer> x = new ArrayList<>();
```

Comment:
As a suggestion, you can make this list like
`List<Integer> x = List.of(1, 2, 3, 4, 5, 6, 7,8,9,1)`